asf_public_discourse_home_decarbonisation/analysis/mse/initial_text_analysis_category_data.py

```diff
@@ -0,0 +1,442 @@
+        keyword_dictionary (dict): Dictionary containing the keywords
+        save_path (str): Path to save the table
+    """
+    logger.info("Creating table with keyword-specific counts...")
+    keyword_counts = pd.DataFrame()
+    keyword_counts["group"] = keyword_dictionary.keys()
+    keyword_counts["counts"] = keyword_counts["group"].apply(
+        lambda x: number_instances_containing_keywords(mse_data, keyword_dictionary[x])
+    )
+    keyword_counts["percentage"] = keyword_counts["counts"] / len(mse_data) * 100
+    keyword_counts.sort_values("counts", ascending=False, inplace=True)
+    logger.info("Keyword groups counts:\n{}".format(keyword_counts))
+
+
+def plot_and_save_analysis_top_n_grams(analysis_parameters: dict):
+    """
+    Plotting and saving the barplots and wordcloud of top ngrams.
+
+    Args:
+        analysis_parameters (dict): A dictionary containing the parameters for the analysis
+    """
+    logger.info("Analysing top keywords and ngrams...")
+    for t in analysis_parameters.keys():
+        params = analysis_parameters[t]
+        filter_to_apply = params["data"]
+        filtered_data = filter_to_apply(mse_data)
+        frequencies = frequency_ngrams(filtered_data, params["ngrams_col"])
+
+        # Bar plot with top ngrams
+        plot_and_save_top_ngrams(
+            frequencies,
+            top_ngrams_barplot,
+            category,
+            params["var_used"],
+            MSE_FIGURES_PATH,
+        )
+
+        # Wordcloud with top ngrams above min_frequency
+        plot_and_save_wordcloud(
+            frequencies,
+            number_ngrams_wordcloud,
+            params["min_frequency"],
+            category,
+            params["var_used"],
+            MSE_FIGURES_PATH,
+            stopwords,
+        )
+
+
+def create_argparser() -> argparse.ArgumentParser:
+    """
+    Creates an argument parser that can receive the following arguments:
+    - category: category or sub-forum (defaults to "green-ethical-moneysaving")
+    - collection_date_time: collection date/time (defaults to "2023_11_15")
+
+    Returns:
+        argparse.ArgumentParser: argument parser
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--category",
+        help="Category or sub-forum",
+        default="green-ethical-moneysaving",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--collection_date_time",
+        help="Collection date/time",
+        default="2023_11_15",
+        type=str,
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+if __name__ == "__main__":
+    args = create_argparser()
+    category = args.category
+    collection_date_time = args.collection_date_time
+
+    MSE_FIGURES_PATH = f"{PROJECT_DIR}/outputs/figures/mse/{category}"
+    os.makedirs(MSE_FIGURES_PATH, exist_ok=True)
+
+    mse_data = get_mse_data(category, collection_date_time)
+
+    stopwords = english_stopwords_definition()
```

Comment:
What's the approach for adding more stopwords as we see fit? This is a more general question in terms of methodology I guess but interested in your view.

Reply:
I don't typically do it with the processing. Instead I do it only after I've taken a look at top words and n-grams. For example, you take a look at the top words/ngrams and you see that you want to remove a few domain specific keywords. So you could pass the domain ones when you plot your word cloud for example. I didn't because nothing too obvious came up. But I had to do it when analysing Twitter data. But i'd separate it from the processing.